frontends/web/src/routes/bitsurance/dashboard.tsx

```diff
@@ -0,0 +1,156 @@
+/**
+ * Copyright 2023 Shift Crypto AG
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useTranslation } from 'react-i18next';
+import { AccountCode, IAccount, getBalance } from '../../api/account';
+import { Button } from '../../components/forms';
+import { alertUser } from '../../components/alert/Alert';
+import { Entry } from '../../components/guide/entry';
+import { Guide } from '../../components/guide/guide';
+import { GuideWrapper, GuidedContent, Header, Main } from '../../components/layout';
+import { View, ViewContent } from '../../components/view/view';
+import style from './dashboard.module.css';
+import { useCallback, useEffect, useState } from 'react';
+import { TAccountDetails, TDetailStatus, bitsuranceLookup } from '../../api/bitsurance';
+import { A } from '../../components/anchor/anchor';
+import { route } from '../../utils/route';
+import { Amount } from '../../components/amount/amount';
+import { useMountedRef } from '../../hooks/mount';
+import { Balances } from '../account/summary/accountssummary';
+import { Skeleton } from '../../components/skeleton/skeleton';
+import { HideAmountsButton } from '../../components/hideamountsbutton/hideamountsbutton';
+import { GreenDot, OrangeDot, RedDot, YellowDot } from '../../components/icon';
+
+
+type TProps = {
+    accounts: IAccount[];
+}
+
+export const BitsuranceDashboard = ({ accounts }: TProps) => {
+  const { t } = useTranslation();
+  const mounted = useMountedRef();
+  const [insuredAccounts, setInsuredAccounts] = useState<TAccountDetails[]>([]);
+  const [balances, setBalances] = useState<Balances>();
+  const [activeAccountCodes, setActiveAccountCodes] = useState<AccountCode[]>([]);
+
+  const accountStatusIcon = (status: TDetailStatus) => {
+    switch (status) {
+    case 'active':
+      return <GreenDot/>;
+    case 'processing':
+    case 'waitpayment':
+      return <YellowDot/>;
+    case 'refused':
+      return <RedDot/>;
+    case 'inactive':
+      return <OrangeDot/>;
+    case 'canceled':
+      return <RedDot/>;
+    }
+  };
```

Comment:
small nit: This is pure function and could be defined before `export const BitsuranceDashboard`, maybe rename to AccountStatusIcon so you can use it in JSX, `<AccountStatusIcon status={account.status}/>`

Reply:
Done!